pkg/app/middleware_test.go

```diff
@@ -73,3 +77,82 @@ var _ = Describe("WithHealthMiddleware", func() {
 		Expect(resp.StatusCode).To(Equal(http.StatusOK))
 	})
 })
+
+var _ = Describe("WithInitrdViaHTTPMiddleware", func() {
+	var (
+		httpServer  *httptest.Server
+		httpsServer *httptest.Server
+		httpClient  *http.Client
+		httpsClient *http.Client
+	)
+
+	BeforeEach(func() {
+		mux := http.NewServeMux()
+		mux.Handle("/", http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			fmt.Fprintln(w, "Hello!")
+		}))
+		httpServer = httptest.NewServer(WithIPXEScriptMiddleware(mux))
+		httpsServer = httptest.NewServer(WithIPXEScriptMiddleware(mux))
```

Comment:
Why is this any different from the http server?
Shouldn't this be `NewTLSServer`?

Reply:
Ah, right, fixed